pkg/virt-controller/watch/preset.go

```diff
@@ -182,7 +185,46 @@ func listPresets(vmiPresetInformer cache.SharedIndexInformer, namespace string)
 		result = append(result, *vmi)
 	})
 
-	return result, err
+	if err != nil {
+		return result, err
+	}
+	return sortPresets(result), nil
```

Comment:
Unfortunately more logic is needed than just sorting. Please refer to the conversation in https://github.com/kubevirt/kubevirt/issues/805. Logic elsewhere in this module ends up defeating this.

In the function `mergeDomainSpec` the first call is to `checkMergeConflicts`. If two presets conflict with each other's settings, the entire set of presets gets rejected. Because only orthogonal presets can currently even be added, ordering doesn't matter. Because we're changing the rules to allow presets to conflict (implied by the fact that we're asserting priority order), you almost certainly want to remove that check and adjust the rules in all the associated tests.

Reply:
Thanks @stu-gott for raising this point, this is exactly the discussion I was hoping to have.

In order to support virt-profiles, conflicts (more precisely, overlaps) should be allowed to some extent.

We can re-evaluate how we handle conflicts between presets - the abstract logic is pretty much the same.
Right now just bail out on overlap: the portion of the spec being updated by the preset should either be blank, or look
like the preset wants it to be. From another angle, the portion of the spec should match exactly what the preset wants.

This means that the preset wants to enforce a "equals to" relation.

To have profiles, we want a "at least to" relation. Few examples:

When applying a profile:
- if the existing specs requests 4 cores and the preset says 2, there is no conflict (4 cores >= 2 :)), so we can go ahead
- if the existing specs requests a cpu model and the preset says another, we can either check if the CPUs are compatible (libvirt has APIs for that, and the check is not host-dependent, we are just comparing abstract models) or raise the conflict; or raise anyway the conflict if the CPUs are not compatible
- if the existing specs has one SPICE device and the preset requires one graphic device, there is no conflict

Reply:
So you're suggesting that what we're going to need is not a one-size-fits all approach to overlaying presets, but more of a fine-grained approach based on what makes sense for each data type. e.g. numeric "minimum requested" sort of logic can apply to number of CPUs but not type.

Reply:
@stu-gott in a nutshell, yes. IMHO *IFF* we want to allow overlaps, *then* we should check  case-by-case. Some things may be allowed, some things cannot.